Tests/TuistDependenciesTests/Mappers/ExternalProjectsPlatformTreeShakerGraphMapperTests.swift

```diff
@@ -0,0 +1,191 @@
+                externalProject.path: [
+                    externalPackage.name: externalPackage,
+                ],
+            ],
+            dependencies: [
+                appTargetDependency: Set([externalPackageDependency]),
+            ]
+        )
+
+        // When
+        let (mappedGraph, _) = try await subject.map(graph: graph)
+
+        // Then
+        XCTAssertEqual(mappedGraph.targets[project.path]![appTarget.name]!.supportedPlatforms, Set([.iOS]))
+        XCTAssertEqual(mappedGraph.targets[externalProject.path]![externalPackage.name]!.supportedPlatforms, Set([.iOS]))
+    }
+
+    func test_map_when_external_transitive_dependency_without_platform_filter() async throws {
+        // Given
+        let directory = try temporaryPath()
+        let packagesDirectory = directory.appending(component: "Dependencies")
+
+        let appTarget = Target.test(name: "App", destinations: [.iPad, .iPhone])
+        let directExternalPackage = Target.test(
+            name: "Direct",
+            destinations: [.iPad, .iPhone],
+            product: .framework
+        )
+        let transitiveExternalPackage = Target.test(
+            name: "Transitive",
+            destinations: [.iPad, .iPhone, .appleWatch, .appleTv, .mac, .macWithiPadDesign, .macCatalyst],
+            product: .framework
+        )
+
+        let project = Project.test(path: directory, targets: [appTarget])
+        let externalProject = Project.test(
+            path: packagesDirectory,
+            targets: [directExternalPackage, transitiveExternalPackage],
+            isExternal: true
+        )
+
+        let appTargetDependency = GraphDependency.target(name: appTarget.name, path: project.path)
+        let directExternalPackageDependency = GraphDependency.target(name: directExternalPackage.name, path: externalProject.path)
+        let transitiveExternalPackageDependency = GraphDependency.target(
+            name: transitiveExternalPackage.name,
+            path: externalProject.path
+        )
+
+        let graph = Graph.test(
+            projects: [
+                directory: project,
+                packagesDirectory: externalProject,
+            ],
+            targets: [
+                project.path: [
+                    appTarget.name: appTarget,
+                ],
+                externalProject.path: [
+                    directExternalPackage.name: directExternalPackage,
+                    transitiveExternalPackage.name: transitiveExternalPackage,
+                ],
+            ],
+            dependencies: [
+                appTargetDependency: Set([directExternalPackageDependency]),
+                directExternalPackageDependency: Set([transitiveExternalPackageDependency]),
+            ]
+        )
+
+        // When
+        let (mappedGraph, _) = try await subject.map(graph: graph)
+
+        // Then
+        XCTAssertEqual(mappedGraph.targets[project.path]![appTarget.name]!.supportedPlatforms, Set([.iOS]))
```

Comment:
Small nit: using `!` can make debugging failures more difficult when `nil` exists unexpectedly since it crashes the whole test suite.  using `XCTUnwrap` will fail the test and not crash the suite and provide more granular insight on what specifically broke. 
(using a suggestion, but github is not an IDE)
```suggestion
        XCTAssertEqual(try XCTUnwrap(mappedGraph.targets[project.path]?[appTarget.name]?.supportedPlatforms), Set([.iOS]))
```

Reply:
Good point! I remember coming across this in the past, but it's been a long time without writing XCTest tests actively, and I forgot all these best principles. Thanks for calling it out.